beluga/test/beluga/sensor/data/test_cache_friendly_grid_storage.cpp

```diff
@@ -0,0 +1,127 @@
+
+namespace beluga {
+
+namespace {
+
+TEST(CacheFriendlyGridStorageTest, SetCellData) {
+  // test that if no content is provided, the grid is empty
+  auto uut = CacheFriendlyGridStorage<int32_t, 16>(50, 50);
+  const auto& const_uut = uut;
+
+  ASSERT_EQ(uut.width(), 50);
+  ASSERT_EQ(uut.height(), 50);
+
+  for (int i = 0; i < static_cast<int>(uut.width()); ++i) {
+    for (int j = 0; j < static_cast<int>(uut.height()); ++j) {
+      ASSERT_EQ(uut.cell(i, j), 0);
+      ASSERT_EQ(const_uut.cell(i, j), 0);
+      uut.cell(i, j) = 99;
+      ASSERT_EQ(uut.cell(i, j), 99);
+      ASSERT_EQ(const_uut.cell(i, j), 99);
+    }
+  }
+}
+
+TEST(CacheFriendlyGridStorageTest, ColoredCacheLinesForSquareArrangement) {
+  using ValueType = int32_t;
+  const std::size_t k_width = 22;
+  const std::size_t k_height = 14;
+  const std::size_t k_line_len = 64;
+  const std::size_t k_tile_size = k_line_len / sizeof(ValueType);
+  const auto k_tile_side = static_cast<std::size_t>(std::sqrt(static_cast<double>(k_tile_size)));
+
+  auto uut = CacheFriendlyGridStorage<ValueType, k_line_len>(k_width, k_height);
+
+  ASSERT_EQ(uut.width(), k_width);
+  ASSERT_EQ(uut.height(), k_height);
+
+  const auto buffer_width = ((k_width + k_tile_side - 1) / k_tile_side) * k_tile_side;
+  const auto buffer_height = ((k_height + k_tile_side - 1) / k_tile_side) * k_tile_side;
+  for (std::size_t i = 0; i < buffer_height * buffer_width; ++i) {
+    uut.raw_data()[i] = static_cast<ValueType>(i / k_tile_size);
+  }
+
+  auto expected_grid_contents = std::vector<ValueType>({
+      0,  0,  0,  0,  1,  1,  1,  1,  2,  2,  2,  2,  3,  3,  3,  3,  4,  4,  4,  4,  5,  5,   //
+      0,  0,  0,  0,  1,  1,  1,  1,  2,  2,  2,  2,  3,  3,  3,  3,  4,  4,  4,  4,  5,  5,   //
+      0,  0,  0,  0,  1,  1,  1,  1,  2,  2,  2,  2,  3,  3,  3,  3,  4,  4,  4,  4,  5,  5,   //
+      0,  0,  0,  0,  1,  1,  1,  1,  2,  2,  2,  2,  3,  3,  3,  3,  4,  4,  4,  4,  5,  5,   //
+      6,  6,  6,  6,  7,  7,  7,  7,  8,  8,  8,  8,  9,  9,  9,  9,  10, 10, 10, 10, 11, 11,  //
+      6,  6,  6,  6,  7,  7,  7,  7,  8,  8,  8,  8,  9,  9,  9,  9,  10, 10, 10, 10, 11, 11,  //
+      6,  6,  6,  6,  7,  7,  7,  7,  8,  8,  8,  8,  9,  9,  9,  9,  10, 10, 10, 10, 11, 11,  //
+      6,  6,  6,  6,  7,  7,  7,  7,  8,  8,  8,  8,  9,  9,  9,  9,  10, 10, 10, 10, 11, 11,  //
+      12, 12, 12, 12, 13, 13, 13, 13, 14, 14, 14, 14, 15, 15, 15, 15, 16, 16, 16, 16, 17, 17,  //
+      12, 12, 12, 12, 13, 13, 13, 13, 14, 14, 14, 14, 15, 15, 15, 15, 16, 16, 16, 16, 17, 17,  //
+      12, 12, 12, 12, 13, 13, 13, 13, 14, 14, 14, 14, 15, 15, 15, 15, 16, 16, 16, 16, 17, 17,  //
+      12, 12, 12, 12, 13, 13, 13, 13, 14, 14, 14, 14, 15, 15, 15, 15, 16, 16, 16, 16, 17, 17,  //
+      18, 18, 18, 18, 19, 19, 19, 19, 20, 20, 20, 20, 21, 21, 21, 21, 22, 22, 22, 22, 23, 23,  //
+      18, 18, 18, 18, 19, 19, 19, 19, 20, 20, 20, 20, 21, 21, 21, 21, 22, 22, 22, 22, 23, 23,  //
+  });
+}
```

Comment:
@glpuga hmm I think we left the assertions out :smiley:

Reply:
Fixed, this an some other initialization weirdness that appeared because default value initialization of the grids got lost along he way when leaving std::vector to use aligned std::unique_ptr to hold buffers.